### PR TITLE
Add del vector pgvector + adding modification time to confluence and google drive docs

### PR DESCRIPTION
**Pull request by NaamaMagami**
Description:
- adding implementation of delete for pgvector
- adding modification time in docs metadata for confluence and google drive.

Issue:
https://github.com/langchain-ai/langchain/issues/9312

Tag maintainer: @baskaryan, @eyurtsev, @hwchase17, @rlancemartin.

### CODE REVIEW

File: libs/langchain/langchain/vectorstores/pgvector.py [hunk 1]
```diff
@@ -168,7 +169,29 @@ def delete_collection(self) -> None:
                 return
             session.delete(collection)
             session.commit()
-
+            
+    def delete(
+        self,
+        ids: Optional[List[str]] = None,
```
Comment (eyurtsev) [inline libs/langchain/langchain/vectorstores/pgvector.py:175]: `ids` should be used to delete `custom_id`  to be consistent with other vectorstores. the `id` is the user assigned id for the document

If we want the ability to delete by `uuid`, we can rename `custom_ids` to `uuids` and have it delete by uuid

**Review by eyurtsev — CHANGES_REQUESTED** [review]
Looks great! One change is to the `delete` method. The `ids` need to correspond to the same ids that the user passes in add_documents

**Review by eyurtsev — APPROVED** [review]

File: libs/langchain/langchain/vectorstores/pgvector.py (hunk 1 above)
Comment (eyurtsev) [reply libs/langchain/langchain/vectorstores/pgvector.py:177]: Thanks! I'm merging and going to make a PR to make the deletion more efficient. Deletion can be made much faster, but using a single SQL request.

File: libs/langchain/langchain/document_loaders/confluence.py [hunk 2]
```diff
@@ -505,13 +505,18 @@ def process_page(
...
             text = text + "".join(comment_texts)
 
+        metadata = {
+            "title": page["title"],
+            "id": page["id"],
+            "source": self.base_url.strip("/") + page["_links"]["webui"],
+        }
+        
+        if "version" in page and "when" in page["version"]:
+            metadata["when"] = page["version"]["when"]
```
Comment (eyurtsev) [inline libs/langchain/langchain/document_loaders/confluence.py:515]: might be better to call it more precisely: `when` -> `updated_at` / `last_modified_at`

to differentiate between update time from creation time

**Comment by eyurtsev** [comment]
@baskaryan feel free to merge whenever

**Comment by NaamaMagami** [comment]
hey, sorry for the delay - no problem, I'm not even sure it's possible to delete by uuid but I thought that maybe some users could use this functionality since the custom_id is not a unique id..
Thanks!

**Pull request merged**